DuckDuckGo/AdAttribution/AdAttributionPixelReporter.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  AdAttributionPixelReporter.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Core
+
+protocol PixelFiring {
+    static func fire(pixel: Pixel.Event, withAdditionalParameters params: [String: String])
+}
```

Comment:
Great to have this - at some point in the future I'd love to expand this and move it into the Pixel.swift file, and encourage people to use it since it makes testability a lot easier. 👏 

We have a proposal kicking around somewhere to add a Pixel module to BSK and include a feature like this, so probably we do it then.